c++/defs/workspace.bzl

```diff
@@ -0,0 +1,14 @@
+"Provides WORKSPACE.bazel initialization macros for Cap'n'Proto and downstream Bazel projects."
+
+load("@bazel_skylib//:workspace.bzl", "bazel_skylib_workspace")
+
+
+def capnp_workspace(omit_skylib = False):
+    """Call initialization macros for a Cap'n'Proto project via a Bazel workspace.
+
+    Args:
+        omit_skylib: Don't initialize skylib.
+    """
+
+    if not omit_skylib:
+        bazel_skylib_workspace()
```

Comment:
This macro, `capnp_workspace`, must be called **only in `WORKSPACE` projects**. Bzlmod projects do not have to load or call this because such initialization is handled automatically by the Bzlmod infrastructure.

Usage example:

**In a `WORKSPACE` file**:
```starlark
load("@capnp-cpp//defs:workspace.bzl", "capnp_workspace")

capnp_workspace()
```